### PR TITLE
Fix/delete required attribute

### PR DESCRIPTION
**Pull request by collinlokken**
## What does this pull request change?
Introduces a check that will not allow deletion of required attributes. This also changes some tests, as a couple of test tries to delete required attributes.

## Issues related to this change:
#516

### CODE REVIEW

File: src/tests/unit/test_remove.py [hunk 1]
```diff
@@ -38,21 +39,39 @@ def test_remove_document(self):
...
+                "type": "dmss://testing/this_blueprint_does_not_exist",
+            }
         }
 
-        class NoBlueprints:
-            def get_blueprint(self, type):
-                raise FileNotFoundError
+        # class NoBlueprints:
+        #    def get_blueprint(self, type):
+        #        raise FileNotFoundError
```
Comment (KristianKjerstad) [inline src/tests/unit/test_remove.py:69]:
```suggestion
```
can be deleted

File: src/tests/unit/test_remove.py [hunk 2]
```diff
@@ -84,9 +103,10 @@ def test_remove_child_dict(self):
             "2": {"name": "Nested", "description": "", "type": "basic_blueprint"},
         }
 
-        self.document_service.remove(Address("$1.nested", "testing"))
-        assert self.storage["1"].get("nested") is None
-        assert self.storage.get("2") is None
+        self.assertRaises(ValidationException, self.document_service.remove, Address("$1.nested", "testing", "dmss"))
+        # TODO Test fails due to modelling error in blueprint. Cannot delete required attribute
```
Comment (KristianKjerstad) [inline src/tests/unit/test_remove.py:107]: would be nice to also add a new issue to the "incoming" tab in the board to fix modeling errors in the blueprints used in this test_remove.py also 👍

**Review by KristianKjerstad — APPROVED** [review]
nice!

File: src/tests/unit/test_remove.py (hunk 2 above)
Comment (collinlokken) [reply src/tests/unit/test_remove.py:107]: Ok, I will add it to the board!

**Pull request merged**